### DISCUSSION TITLE
Version 1.0.1

### DISCUSSION THREAD

**Discussion by ZZ-Cat** [ZZ-Cat/CRSFforArduino#96] [Announcements]
## Overview

This is a follow-up release to Version 1.0.0, and it is for those that were unable to install or build CRSF for Arduino in the Arduino IDE.

### Fixes

- fix(arduino): Arduino IDE build failure by @ZZ-Cat in https://github.com/ZZ-Cat/CRSFforArduino/pull/94

### Chores

- ci(workflows): Arduino, Auto-close, and PlatformIO maintenance by @ZZ-Cat in https://github.com/ZZ-Cat/CRSFforArduino/pull/91

### More information

**Full Changelog**: https://github.com/ZZ-Cat/CRSFforArduino/compare/v1.0.0...v1.0.1

---

## Semantic Versioning

All releases of CRSF for Arduino follow the [Semantic Versioning Guidelines.](https://semver.org/)
This means "Major.Minor.Patch".
Major = New features that are incompatible with previous iterations.
Minor = New features &/or hardware support that retains compatibility with the current Major release.
Patch = Bug fixes.

This is Patch Version is the first follow-up release to Major Version 1.x.x.
While every effort has gone in to ensure the latest release is bug-free, there are no guarantees with firmware development.
Your feedback is always welcome. If you spot any bugs, don't hesitate to open an [issue.](../../issues)
For general comments & questions, please use the [discussions](../../discussions) section.
New contributors are always welcome. Please read through my [Contribution Guidelines](../Main-Trunk/.github/CONTRIBUTING.md) before you go Gung Ho with coding.

---

### Arduino IDE v2.x.x

1. Download the `Source Code (ZIP)` in the `Assets` section of this Release, and save to a convenient space on your hard drive.
2. In the Arduino IDE, go to `Sketch` -> `Include Library` -> `Add .ZIP Library...`
3. Click `CRSFforArduino-1.0.1` and Arduino IDE will automatically install it for you.
4. Build-test one of the examples. preferably the `rc_channels` example by `File` -> `Examples` -> `CRSFforArduino` -> `rc_channels`.
5. Click `Verify` and the build should pass.

### PlatformIO

Unless you have pinned CRSF for Arduino's version, you need not do anything, as PlatformIO will automatically update your local copy for you on the next time you build your project's firmware.

<hr /><em>This discussion was created from the release <a href='https://github.com/ZZ-Cat/CRSFforArduino/releases/tag/v1.0.1'>Version 1.0.1</a>.</em>

**Comment by SB-JHL-B26** [comment]
1) The above Arduino IDE v2.x.x instructions to not work using the Add .ZIP Library function. It does work by extracting the ZIP and manually moving the CRSFforArduino-1.0.1 folder to the ".../Arduino/libraries" folder. 

2) The code does compile now. However, there is no output for the rc_channels example. there is no serial monitor output such as "CRSF for Arduino initialisation failed!" or "RC Channels Example" or "Ready" as you might expect by reading the example code.

3) There is no Serial1.begin such as  Serial1.begin(420000, SERIAL_8N1, D0, D1); in the setup() and no comments in the code where to configure the Rx/Tx Serial1 pins. I could not find any documentation that provides information to set up the board UART that is wired to the ELRS Rx.  

I am using an Arduino Nano ESP32 connected via a USB port. My hardware setup works perfectly right out of the box with the AlfredoCRSF library once the Rx and Tx  pins are changed to "crsfSerial.begin(CRSF_BAUDRATE, SERIAL_8N1, D0, D1);". This line of code is right in the setup() and easy to spot.

I am using a RadioMaster TX16S with a RadioMaster RP 2 Rx. My hardware test setup work perfectly with the AlfredoCRSF library.

At this point, I do not know what else I can do to help. What is required is more pre-release testing to ensure that a user can simple load a library and run an example and have it work right out of the box. Unfortunately, this has not been my experience. I read similar comments from another thread. 

At some point down the road I will pick this up and do some more testing again should there be a 1.1 release.

Cheers, s

**Reply by ZZ-Cat** [reply]
Would you mind transferring this to my Issues tab as a bug report?
